### PR TITLE
Fixed missing signaling of io error

### PR DESCRIPTION
**Pull request by KimBoendergaard**
Catch closed TLS connections

### CODE REVIEW

File: port/src/tlsio_esp_tls.c [hunk 1]
```diff
@@ -484,19 +491,18 @@ static void tlsio_esp_tls_dowork(CONCRETE_IO_HANDLE tls_io)
...
             if (dowork_read(tls_io_instance) < 0 && errno != EAGAIN)
             {
-                tls_io_instance->tlsio_state = TLSIO_STATE_ERROR;
+                enter_tlsio_error_state(tls_io_instance);
             }
             if (dowork_send(tls_io_instance) < 0 && errno != EAGAIN)
             {
-                tls_io_instance->tlsio_state = TLSIO_STATE_ERROR;
-            }
+                enter_tlsio_error_state(tls_io_instance);            }
```
Comment (mogenslu) [inline port/src/tlsio_esp_tls.c]: Missing a newline

**Review by mogenslu — APPROVED** [review]

**Review by HugoChristensen — APPROVED** [review]
Do not have the overview to see if this is the mem leak problem. But I don't think it introduce new errors

File: port/src/tlsio_esp_tls.c (hunk 1 above)
Comment (KimBoendergaard) [reply port/src/tlsio_esp_tls.c]: Fixed

**Comment by KimBoendergaard** [comment]
> Do not have the overview to see if this is the mem leak problem. But I don't think it introduce new errors

Well - it doesn't really have to do with memory leak. It was just an issue I found while testing my memory leak 'fixes'.
This fix ensure MQTT and thus CloudMgr is notified if the underlying TLS can't reach its peer anymore (due to e.g. loss of Wi-Fi connection)

**Pull request merged**